File: LC-392/LC-392-3.cpp
class Solution {
public:
    /**
     * Concepts: Recursive
     */
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) return true;
        
        int tPtr = t.find(s[0]);
        if (tPtr > -1)
            return isSubsequence(s.substr(1), t.substr(tPtr + 1));
        return false;
    }
};

File: LC-392/LC-392-3.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  /**
   * Concepts: Recursive
   */

  if (s.length == 0) return true

  let tPtr = t.indexOf(s[0])
  if (tPtr > -1) return isSubsequence(s.substr(1), t.substr(tPtr + 1))
  return false
}
